### PR TITLE
EVG-15355 continue adding project admins on error

### PR DESCRIPTION
**Pull request by bynn**
[EVG-15355](https://jira.mongodb.org/browse/EVG-15355)

### Description 
does not return on error while adding admins

### Testing 
unit test

### CODE REVIEW

File: model/project_ref.go [hunk 1]
```diff
@@ -1885,23 +1885,31 @@ func (p *ProjectRef) UpdateAdminRoles(toAdd, toRemove []string) error {
...
 		if viewRole != "" {
 			if err = adminUser.AddRole(viewRole); err != nil {
-				return errors.Wrapf(err, "error adding role %s to user %s", viewRole, addedUser)
+				catcher.Wrapf(err, "error adding role %s to user %s", viewRole, addedUser)
+				continue
 			}
 		}
 	}
+	if err = catcher.Resolve(); err != nil {
+		return errors.Wrap(err, "error adding users")
```
Comment (ablack12) [inline model/project_ref.go:1911]: let's say "error adding some users" since any users that aren't in this error _were_ added

**Review by ablack12 — CHANGES_REQUESTED** [review]
We'll want to extend this change to the toRemove users below, and also to the repo ref method: https://github.com/evergreen-ci/evergreen/blob/14bc0f57bbb164f4377bb26e767fdfcde52953a5/model/repo_ref.go#L234

File: model/project_ref.go (hunk 1 above)
Comment (bynn) [reply model/project_ref.go]: done

**Review by ablack12 — CHANGES_REQUESTED** [review]
Hi sorry to make this more complicated (and theoretically this was actually already wrong): if we're adding/removing roles for some users, we _should_ update the existing admin list (see https://github.com/evergreen-ci/evergreen/blob/dc484d82582b2fd4dc151054a2d06c41fc9fb077/rest/data/project.go#L142: we want to get down to SaveProjectPageForSection so we save the admin list, and then log the project as modified), even though we errored for some users.

This means that we have UpdateAminRoles should probably remove from p.Admins any users it failed to add, and similarly add to p.Admins any users it failed to remove. Then we can continue on error in SaveProjectSettingsForSection and save the admins list in the state that it currently is in. Let me know if you have questions with this.

**Review by ablack12 — APPROVED** [review]
This looks good and we can fix the other discrepancies we discussed in the future ticket

**Pull request merged**